### ISSUE TITLE
NaN loss in time series use case

### ISSUE THREAD

**Issue by paxcema** [mindsdb/lightwood#350]
**Versions**
* Lightwood 0.54
* MindsDB Native 2.29.0

** Bug **
Using `metro_traffic_ts` dataset with the following `learn()` call:

```
mdb.learn(from_data='metro_traffic_ts.csv',
          to_predict='traffic_volume',
          timeseries_settings={'order_by': ['date_time'],
                               'window': 3
                               },
          advanced_args={'force_predict': True,
                         'disable_column_importance': True })
```

Results in the following behavior soon after training starts:

```
...
Got weird target value to decode: [nan, nan, nan]
Got weird target value to decode: [nan, nan, nan]
DEBUG:mindsdb-logger-5f549222-49d9-11eb-9355-4116bc1fb6ed---no_report:/MindsDB/mindsdb_native/mindsdb_native/libs/phases/model_interface/lightwood_backend.py:280 - We've reached training epoch nr 148 with an r2 score of -2.430622829048794e+119 on the testing dataset
```

**Comment by paxcema** [comment]
The problem here is caused by AMP usage in the NnMixer without a gradient scaler, which can result in underflows as seen in the issue. #349 introduces `GradScaler` usage which solves this without compromising on performance or accuracy.

**Comment by paxcema** [comment]
Addressed in #380
